Brokerages/Binance/BinanceBrokerage.Messaging.cs

```diff
@@ -0,0 +1,317 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using QuantConnect.Brokerages.Binance.Messages;
+using QuantConnect.Data.Market;
+using QuantConnect.Logging;
+using QuantConnect.Orders;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Securities;
+using RestSharp;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Net;
+
+namespace QuantConnect.Brokerages.Binance
+{
+    public partial class BinanceBrokerage
```

Comment:
style - I never was a fan of the partial class pattern for brokerages. My preference would be to see multiple smaller classes instead of a single master class. The partial class pattern has been used in others, so we don't _need_ to change this as it matches existing patterns, but I do think it's worthwhile thinking about whether or not it's the best pattern. I could see what is currently a single brokerage being split into (for example) a `BinanceApiClient` to manage the request/response between lean and binance and a `BinanceBrokerage` that would use the `BinanceApiClient` as an adapter for the `IBrokerage` interface. This provides a nice separation of concerns and makes other components available for reuse.